src/wpsmon.c

```diff
@@ -329,7 +329,12 @@ void monitor(char *bssid, int passive, int source, int channel, int mode)
 		{
         		act.sa_handler = sigalrm_handler;
         		sigaction (SIGALRM, &act, 0);
-			ualarm(CHANNEL_INTERVAL, CHANNEL_INTERVAL);
+			struct itimerval timer;
```

Comment:
this can be simplified to `struct itimerval timeroff = {.it_value.tv_usec = xxx, .it_interval.tv_usec = xxx};`